plenoptic/simulate/models/frontend.py

```diff
@@ -1,96 +1,407 @@
+            kernel_size,
+            center,
+            width_ratio_limit,
+            amplitude_ratio,
+            pad_mode=pad_mode,
+        )
+        self.activation = activation
 
-        """
+    def forward(self, x: Tensor) -> Tensor:
+        y = self.activation(self.center_surround(x))
+        return y
+
+
+class LG(nn.Module):
+    """ Linear center-surround followed by luminance gain control and activation.
+    Parameters
+    ----------
+    See `CenterSurround` class for full Parameter docstring.
+
+    Attributes
+    ----------
+    center_surround: nn.Module
+        Difference of Gaussians linear filter.
+    luminance: nn.Module
+        Gaussian convolutional kernel used to normalize signal by local luminance.
+    luminance_scalar: nn.Parameter
+        Scale factor for luminance normalization.
+    """
+    def __init__(
+        self,
+        kernel_size: Union[int, Tuple[int, int]],
+        center: str = "on",
+        width_ratio_limit: float = 4.0,
+        amplitude_ratio: float = 1.25,
+        pad_mode: str = "circular",
+        activation: Callable[[Tensor], Tensor] = F.softplus,
+    ):
         super().__init__()
+        self.center_surround = CenterSurround(
+            kernel_size,
+            center,
+            width_ratio_limit,
+            amplitude_ratio,
+            pad_mode=pad_mode,
+        )
+        self.luminance = Gaussian(kernel_size=kernel_size)
+        self.luminance_scalar = nn.Parameter(torch.rand(1) * 10)
+        self.activation = activation
 
-        # convolutional weights
-        self.linear = nn.Conv2d(in_channels=1, out_channels=2, kernel_size=31, bias=False)
-        self.luminance = nn.Conv2d(in_channels=1, out_channels=2, kernel_size=31, bias=False)
-        self.contrast = nn.Conv2d(in_channels=2, out_channels=2, kernel_size=31, groups=2, bias=False)
+    def forward(self, x: Tensor) -> Tensor:
+        linear = self.center_surround(x)
+        lum = self.luminance(x)
+        lum_normed = linear / (1 + self.luminance_scalar * lum)
+        y = self.activation(lum_normed)
+        return y
 
-        # contrast and luminance normalization scaling
-        self.luminance_scalars = nn.Parameter(torch.rand((1, 2, 1, 1)))
-        self.contrast_scalars = nn.Parameter(torch.rand((1, 2, 1, 1)))
 
-        # pad all transforms for convolution
-        pad = nn.ReflectionPad2d(self.linear.weight.shape[-1]//2)
-        self.linear_pad = transforms.Compose([pad, self.linear])
-        self.luminance_pad = transforms.Compose([pad, self.luminance])
-        self.contrast_pad = transforms.Compose([pad, self.contrast])
-        self.softplus = nn.Softplus()
+class LGG(nn.Module):
+    """ Linear center-surround followed by luminance and contrast gain control,
+    and activation function.
 
-        self.disk_mask = disk_mask
-        self._disk = None  # cached disk to apply to image
+    Parameters
+    ----------
+    See `CenterSurround` class for full Parameter docstring.
```

Comment:
Seems like a good opportunity to try out something from #25: have the parameters docstring be a constant that gets reused across classes.